app/controllers/application_controller.rb

```diff
@@ -62,6 +62,18 @@ def params_for_update
     body_params.permit(*api_doc_definition.all_attributes - api_doc_definition.read_only_attributes)
   end
 
+  def request_path
+    request.env["REQUEST_URI"]
+  end
+
+  def request_path_parts
+    request_path.match(/\/(?<full_version_string>v\d+.\d+)\/(?<primary_collection_name>\w+)\/?(?<primary_collection_id>\d+)?\/?(?<subcollection_name>\w+)?/) || {}
+  end
+
+  def allowed_params_for_primary_collection
+    ["#{request_path_parts[:primary_collection_name].singularize}_id", "#{request_path_parts[:primary_collection_name].singularize}_type"]
```

Comment:
@bdunne when I test this with `sources/availabilities` (https://github.com/ManageIQ/topological_inventory-api/pull/127/commits/49f52fd02a0aca4b248762b5ec5baa88ec44848d) this results in `["source_id", "source_type"]` when it would have to be `["resource_id", "resource_type"]` with `resource_type = Source` to work.